Sources/SwiftDocC/Model/Rendering/Content/RenderBlockContent+Capitalization.swift

```diff
@@ -8,47 +8,50 @@
  See https://swift.org/CONTRIBUTORS.txt for Swift project authors
 */
 
-/// For auto capitalizing the first letter of a sentence following a colon (e.g. asides, sections such as parameters, returns).
-protocol AutoCapitalizable {
-    
-    /// Any type that conforms to the AutoCapitalizable protocol will have the first letter of the first word capitalized (if applicable).
-    var withFirstWordCapitalized: Self {
-        get
-    }
-    
-}
-
-extension AutoCapitalizable {
-    var withFirstWordCapitalized: Self { return self }
-}
 
-extension RenderInlineContent: AutoCapitalizable {
+extension RenderInlineContent {
     /// Capitalize the first word for normal text content, as well as content that has emphasis or strong applied.
-    var withFirstWordCapitalized: Self {
+    func capitalizeFirstWord() -> Self {
```

Comment:
I think the common way to name this function in Swift would be "capitaliz**ing**FirstWord()".

```suggestion
    func capitalizingFirstWord() -> Self {
```

Because this is a non-mutating function one would normally strive for a noun phrase but I have a hard time thinking of one that works well. Instead, I feel that an "ing" suffix on the verb is the most readable naming here.

See this section of the [Swift API Design Guidelines][1]:

[1]: https://www.swift.org/documentation/api-design-guidelines/


> **Name functions and methods according to their side-effects**
>
> - Those without side-effects should read as noun phrases, e.g. `x.distance(to: y)`, `i.successor()`.
>
> - Those with side-effects should read as imperative verb phrases, e.g., `print(x)`, `x.sort()`, `x.append(y)`.
>
> - **Name Mutating/nonmutating method pairs consistently.** A mutating method will often have a nonmutating variant with similar semantics, but that returns a new value rather than updating an instance in-place.
>
>   - When the operation is naturally described by a verb, use the verb’s imperative for the mutating method and apply the “ed” or “ing” suffix to name its nonmutating counterpart.
>     - Prefer to name the nonmutating variant using the verb’s past [participle](https://en.wikipedia.org/wiki/Participle) (usually appending “ed”):
>       ` ... `
>     - When adding “ed” is not grammatical because the verb has a direct object, name the nonmutating variant using the verb’s present [participle](https://en.wikipedia.org/wiki/Participle), by appending “ing.”
>       ```
>       /// Strips all the newlines from `self`
>       mutating func stripNewlines()
>
>       /// Returns a copy of `self` with all the newlines stripped.
>       func strippingNewlines() -> String
>       ...
>       ```

Reply:
Sounds great, thanks for adding the description! Makes it very clear and I think this makes a lot of sense as well.